src/number_theory/arithmetic_function.lean

```diff
@@ -530,6 +534,116 @@ begin
+  ω n = (unique_factorization_monoid.factors n).to_finset.card :=
+begin
+  rw [nat.factors_eq, finset.card, multiset.to_finset_val,
+  multiset.coe_erase_dup, multiset.coe_card],
+  refl
+end
+
+/-- `μ` is the Möbius function. If `n` is squarefree with an even number of distinct prime factors,
+  `μ n = 1`. If `n` is squarefree with an odd number of distinct prime factors, `μ n = -1`.
+  If `n` is not squarefree, `μ n = 0`. -/
+def moebius : arithmetic_function ℤ :=
+⟨λ n, if squarefree n then (-1) ^ (card_factors n) else 0, by simp⟩
+
+localized "notation `μ` := moebius" in arithmetic_function
+
+@[simp]
+lemma moebius_one : μ 1 = 1 :=
+by simp [moebius]
+
+@[simp]
+lemma moebius_apply_of_squarefree {n : ℕ} (h : squarefree n): μ n = (-1) ^ (card_factors n) :=
+if_pos h
+
+@[simp]
+lemma moebius_eq_zero_of_not_squarefree {n : ℕ} (h : ¬ squarefree n): μ n = 0 := if_neg h
+
+lemma moebius_ne_zero_iff_squarefree {n : ℕ} : μ n ≠ 0 ↔ squarefree n :=
+begin
+  refine ⟨_, λ h, _⟩,
+  { contrapose!,
+    intro h,
+    exact if_neg h },
+  { rw moebius_apply_of_squarefree h,
+    apply pow_ne_zero,
+    simp }
+end
+
+lemma moebius_ne_zero_iff_eq_or {n : ℕ} : μ n ≠ 0 ↔ μ n = 1 ∨ μ n = -1 :=
+begin
+  split; intro h,
+  { rw moebius_ne_zero_iff_squarefree at h,
+    rw moebius_apply_of_squarefree h,
+    apply neg_one_pow_eq_or },
+  { rcases h with h | h; simp [h] }
+end
+
+open unique_factorization_monoid
+
+/-- Moebius Inversion -/
+lemma moebius_mul_zeta_eq_one : μ * ζ = 1 :=
+begin
+  ext x,
+  cases x, simp,
+  cases x, simp,
+  rw [mul_zeta_apply, one_apply_ne (ne_of_gt (succ_lt_succ (nat.succ_pos _)))],
+  rw ← sum_filter_ne_zero,
+  simp only [moebius_ne_zero_iff_squarefree],
+  transitivity,
+  convert (sum_divisors_filter_squarefree (nat.succ_ne_zero _)),
+  apply eq.trans (sum_congr rfl _) (sum_powerset_neg_one_pow_card_of_nonempty _),
+  { intros y hy,
+    rw [finset.mem_powerset, ← finset.val_le_iff, multiset.to_finset_val] at hy,
+    have h : unique_factorization_monoid.factors y.val.prod = y.val,
+    { apply factors_multiset_prod_of_irreducible,
+      intros z hz,
+      apply irreducible_of_factor _ (multiset.subset_of_le
+        (le_trans hy (multiset.erase_dup_le _)) hz) },
+    rw [if_pos],
+    { rw [card_factors_apply, h, finset.card] },
+    rw [unique_factorization_monoid.squarefree_iff_nodup_factors, h],
+    { apply y.nodup },
+    rw [ne.def, multiset.prod_eq_zero_iff],
+    intro con,
+    rw ← h at con,
+    exact not_irreducible_zero (irreducible_of_factor 0 con) },
+  { apply_instance },
+  { rw finset.nonempty,
+    rcases wf_dvd_monoid.exists_irreducible_factor _ (nat.succ_ne_zero _) with ⟨i, hi⟩,
+    { rcases exists_mem_factors_of_dvd (nat.succ_ne_zero _) hi.1 hi.2 with ⟨j, hj, hj2⟩,
+      use j,
+      apply multiset.mem_to_finset.2 hj },
+    rw nat.is_unit_iff,
+    omega },
+end
+
+lemma zeta_mul_moebius_eq_one : ζ * μ = 1 :=
+by rw [mul_comm, moebius_mul_zeta_eq_one]
+
```

Comment:
Tempting to use https://leanprover-community.github.io/mathlib_docs/algebra/invertible.html#invertible, and add something like

```
instance : invertible ζ :=
{ inv_of := μ,
  inv_of_mul_self := moebius_mul_zeta_eq_one,
  mul_inv_of_self := zeta_mul_moebius_eq_one}
```

Reply:
Yeah, I could do this, or `is_unit`, but there's actually a bigger theorem that an arithmetic function is a unit if and only if its value at 1 is a unit, and it may well be slicker just to use that.